src/uct/base/uct_iface.h

```diff
@@ -678,32 +678,81 @@ size_t uct_iov_total_length(const uct_iov_t *iov, size_t iovcnt)
 }
 
 /**
- * Fill iovec data structure by data provided in uct_iov_t.
+ * Update an array of UCT IOV elements to consider an already consumed data.
+ *
+ * @param [in]     iov            A pointer to an array of UCT IOV elements.
+ * @param [in]     iov_cnt        A number of elements in a iov array.
+ * @param [in/out] cur_iov_idx    A pointer to an index in a iov array from
+ *                                which the operation should be started.
+ * @param [in]     consumed       An amount of data consumed that should be
+ *                                considered in a current iov array.
+ */
+static UCS_F_ALWAYS_INLINE
+void uct_iov_advance(uct_iov_t *iov, size_t iovcnt,
```

Comment:
where is this function used?
also, can we add new file "uct_iov.inl"?

Reply:
it will be used by SCOPY iface in progress function to complete RMA operation for a segment and advance user's IOV, to start RMA operation for the next segment from the correct IOV index and offset in the IOV element